.github/workflows/ci.yaml

```diff
@@ -8,25 +8,23 @@ on:
       - main
 
 jobs:
-  lint:
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [ 'ubuntu-22.04' ]
-
-    name: ${{ matrix.os }} / Lint
-    runs-on: ${{ matrix.os }}
+  # It's recommended to run golangci-lint in a job separate from other jobs (go test, etc) because different jobs run in parallel.
+  go-linter:
+    runs-on: ubuntu-latest
```

Comment:
Would we want to run the lint on Windows as well?

Reply:
I think as a langage level linter, go-linter may not be useful to run in multiple platforms. The impact is only code style without any platform-specific outputs. Code style change should not impact any binaries and functionalities.

Reply:
Oh right, static analysis shouldn't matter the platform. Sounds good.